test/Spec.hs

```diff
@@ -150,7 +152,65 @@ processTicketSpec =
                         zendeskResponses <- run appExecution
 
                         -- Check we have some comments.
-                        assert $ length zendeskResponses > 0
+                        assert $ not (null zendeskResponses)
+
+        it "processes ticket, with no attachments" $
+            forAll (listOf1 genCommentWithNoAttachment) $ \(commentsWithoutAttachment :: [Comment]) ->
+                forAll arbitrary $ \ticketInfo ->
+                    monadicIO $ do
+
+                    let stubbedZendeskLayer :: ZendeskLayer App
+                        stubbedZendeskLayer =
+                            emptyZendeskLayer
+                                { zlGetTicketComments = \_ -> pure commentsWithoutAttachment
+                                , zlGetTicketInfo     = \_ -> pure ticketInfo
+                                , zlPostTicketComment = \_ -> pure ()
+                                }
+
+                    let stubbedConfig :: Config
+                        stubbedConfig = withStubbedIOAndZendeskLayer stubbedZendeskLayer
+
+                    let appExecution :: IO [ZendeskResponse]
+                        appExecution = runApp (processTicket . ticketId $ ticketInfo) stubbedConfig
+
+                    zendeskResponses <- run appExecution
+
+                    assert $ not (null zendeskResponses) && isTaggedWithNoLogs zendeskResponses
+
+        it "processes ticket, with attachments" $
+            forAll (listOf1 arbitrary) $ \(comments:: [Comment]) ->
+                forAll arbitrary $ \ticketInfo ->
+                    monadicIO $ do
+
+                    pre $ any (not . null . cAttachments) comments
+
+                    let stubbedZendeskLayer :: ZendeskLayer App
+                        stubbedZendeskLayer =
+                            emptyZendeskLayer
+                                { zlGetTicketComments = \_ -> pure comments
+                                , zlGetTicketInfo     = \_ -> pure ticketInfo
+                                , zlPostTicketComment = \_ -> pure ()
+                                , zlGetAttachment     = \_ -> pure mempty
+                                }
+
+                    let stubbedConfig :: Config
+                        stubbedConfig = withStubbedIOAndZendeskLayer stubbedZendeskLayer
+
+                    let appExecution :: IO [ZendeskResponse]
+                        appExecution = runApp (processTicket . ticketId $ ticketInfo) stubbedConfig
+
+                    zendeskResponses <- run appExecution
+                    assert $ not (null zendeskResponses) && not (isTaggedWithNoLogs zendeskResponses)
+
+isTaggedWithNoLogs :: [ZendeskResponse] -> Bool
```

Comment:
Can you insert the param name? I know this is also a convention, but I **have to** look at the type to know what this is doing, which is not helping readability.
Try to be as explicit as possible, it helps readability and maintainability.